src/openforms/submissions/logic/rules.py

```diff
@@ -0,0 +1,163 @@
+        if current_step
+        else -1
+    )
+
+    return [
+        rule
+        for rule in rules
+        if _include_rule(submission_state.form_steps, rule, step_index)
+    ]
+
+
+def get_current_step(submission: Submission) -> Optional[SubmissionStep]:
+    """
+    Obtain what the 'current step' of a submission is.
+
+    The current step is defined as the step following the last completed step. Note
+    that this does not evaluate any logic, so if this step happens to be made
+    not-applicable dynamically, it will still be returned.
+
+    If the last completed step is the last step in the form, then this step is returned.
+
+    :arg submission: The submission instance to get the current step for.
+    :returns: None if there are no steps in the form, otherwise the first submission
+      step that isn't completed.
+    """
+    submission_state = submission.load_execution_state()
+    if not submission_state.form_steps:
+        return
+
+    last_completed_step = submission_state.get_last_completed_step()
+    # no steps completed -> current index is first step: 0
+    if last_completed_step is None:
+        current_index = 0
+    else:
+        last_completed_step_index = submission_state.form_steps.index(
+            last_completed_step.form_step
+        )
+        current_index = min(
+            last_completed_step_index + 1, len(submission_state.form_steps) - 1
+        )
+
+    step = submission_state.submission_steps[current_index]
+    return step
+
+
+@dataclass()
+class EvaluatedRule:
+    rule: FormLogic
+    triggered: bool
+
+
+def iter_evaluate_rules(
+    rules: Iterable[FormLogic],
+    data_container: DataContainer,
+    on_rule_check: Optional[Callable[[EvaluatedRule], None]] = None,
+) -> Iterator[ActionOperation]:
+    """
+    Iterate over the rules and evaluate the trigger, yielding action operations.
+
+    Every rule trigger is checked, and if the trigger evaluates to be truthy, all
+    actions in the rule are compiled, yielding action operations for each of them. Any
+    action operator that updates a variable is processed immediately. The caller is
+    responsible for processing (all other) actions accordingly.
+
+    :arg rules: An iterable of form logic rules to evaluate.
+    :arg data_container: The :class:`DataContainer` instance wrapping the
+      submission/step data and everything contained within. Note that the internal state
+      can and should be mutated while processing the action operations (e.g. when updating
+      variable values).
+    :arg on_rule_check: Optional callable taking a :class:`EvaluatedRule` instance as
+      sole argument. Useful to gather metadata about rule evaluation.
+    :returns: An iterator yielding :class:`ActionOperation` instances.
+    """
+    for rule in rules:
+        with elasticapm.capture_span(
+            "evaluate_rule",
+            span_type="app.submissions.logic",
+            labels={"ruleId": rule.pk},
+        ):
```

Comment:
This should result in:

* getting timings of the evaluation of a single rule
* get some aggregated insights of how many rules are evaluated within a transaction (APM should say "x number of calls" or something like that in the UI)